content/deyemiobaa/creating-queues-using-javascript.md

```diff
@@ -0,0 +1,206 @@
+
+## Creating a queue with its operations 
+
+The first step is to initialize our class with its own storage (an array where our queue elements would be stored):
+
+```js
+class Queue {
+  constructor() {
+    this.queue = [];
+  }
+}
+```
+
+The code block above represents the creation of a class, the [`constructor()`](https://www.codecademy.com/resources/docs/javascript/constructors) method is a special method of a class that used to create an object instance of that class. It can be used to initialize any object that would be used across the class methods. The [`this`](https://www.codecademy.com/resources/docs/javascript/this) keyword serves as a regular object within this context. It can be used to initialize any value.
+
+Next up, we add each queue operation as a method of the class.
+
+### Enqueue
+
+This term refers to adding a new element to the queue. Since we're implementing our queue with an array, we can use the `.push()` array method to add new elements to the queue.
+
+```js
+class Queue {
+  constructor() {
+    this.queue = [];
+  }
+
+  enqueue(element) {
+    this.queue.push(element);
+  }
+}
+```
+
+The added lines to our code block represnt a method of class `Queue`. it handles one operation, adding a new item to the array object initialized in our constructor.
+The method accepts one argument `element` and then adds it to `this.queue` using the `.push()` array method.
+
+### Dequeue
+
+This term refers to removing a new element from the queue. Again, the `.shift()` array method takes care of this for us easily.
+
+```js
+class Queue {
+  constructor() {
+    this.queue = [];
+  }
+
+  enqueue(element) {
+    this.queue.push(element);
+  }
+
+  dequeue() {
+    return this.queue.shift();
+  }
+}
+```
+
+We added a new method `dequeue()`. This method deoesn't accept any arguments unlike the previous method. It returns an element from the front of the queue and also removes it using the `.shift()` array method.
+
+### Peek
+
+We use the peek method to check for the element at the front of the queue, without removing it:
+
+```js
+class Queue {
+  constructor() {
+    this.queue = [];
+  }
+
+  enqueue(element) {
+    this.queue.push(element);
+  }
+
+  dequeue() {
+    return this.queue.shift();
+  }
+
+  peek() {
+    return this.queue[0];
+  }
+}
+```
+
+The `peek()` method checks for the value at the front of the queue by accessing the first index of the queue.
+
+### Reversing a Queue
+
+As the title implies, we are simply trying to change the order of the queue from back to front.
+
+```js
+class Queue {
+  constructor() {
+    this.queue = [];
+  }
+
+  enqueue(element) {
+    this.queue.push(element);
+    return this.queue;
+  }
+
+  dequeue() {
+    return this.queue.shift();
+  }
+
+  peek() {
+    return this.queue[0];
+  }
+
+  reverse() {
+    const reversed = []
+    while (this.queue.length > 0) {
+      reversed.push(this.queue.pop());
+    }
+    this.queue = reversed;
+    return this.queue;
+  }
+}
+```
+
```

Comment:
I advise taking out lines 170-174. Place line 170 at the end of 138 and use the bullet points as comments within the snippet. I've provided examples of what this would look like above.